libraries/stdlib/src/kotlin/collections/Collections.kt

```diff
@@ -434,6 +434,35 @@ public fun <T> List<T>.binarySearch(fromIndex: Int = 0, toIndex: Int = size, com
     return -(low + 1)  // key not found
 }
 
+/**
+ * If this list starts with the given [prefix], returns a view of the portion of this list without [prefix].
+ * The returned list is backed by this list, so non-structural changes in the returned list are reflected in this list, and vice-versa.
+ *
+ * Structural changes in the base list make the behavior of the view undefined.
+ */
+public fun <T> List<T>.removePrefix(prefix: List<T>): List<T> {
+    return if (this.startWith(prefix)) {
+        this.subList(prefix.size, this.size)
+    } else {
+        this
+    }
+}
+
+/**
+ * Returns `true` if this list starts with the specified prefix.
+ *
+ * For empty [prefix] returns true
+ */
+public fun <T> List<T>.startWith(prefix: List<T>): Boolean {
```

Comment:
Shouldn't this be "startsWith", to be consistent with the current implementation for String?